crates/stages/src/stages/tx_lookup.rs

```diff
@@ -0,0 +1,295 @@
+use crate::{
+    db::Transaction, exec_or_return, ExecAction, ExecInput, ExecOutput, Stage, StageError, StageId,
+    UnwindInput, UnwindOutput,
+};
+use reth_db::{
+    cursor::{DbCursorRO, DbCursorRW},
+    database::Database,
+    tables,
+    transaction::{DbTx, DbTxMut},
+};
+use tracing::*;
+
+const TRANSACTION_LOOKUP: StageId = StageId("TransactionLookup");
+
+/// The transaction lookup stage.
+///
+/// This stage walks over the bodies table, and sets the transaction hash of each transaction in a
+/// block to the corresponding `TransitionId` at each block. This is written to the
+/// [`tables::TxHashNumber`] This is used for looking up changesets via the transaction hash.
+#[derive(Debug)]
+pub struct TransactionLookupStage {
+    /// The number of table entries to commit at once
+    pub commit_threshold: u64,
+}
```

Comment:
nit: would rather have a `new` function instead of init struct via public, so `StageBuilder` will be consistent with how it creates the stages